core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -185,11 +187,11 @@ class TransactionCoordinator(brokerId: Int,
     } else {
       // caller should have synchronized on txnMetadata already
       txnMetadata.state match {
-        case PrepareAbort | PrepareCommit =>
+        case PrepareAbort | PrepareCommit | Dead =>
```

Comment:
ok - i think you are saying leave it in `pendingState` of `Dead` and remove it from the cache?